analyzers/src/SonarAnalyzer.CSharp/Rules/MethodShouldBeNamedAccordingToSynchronicity.cs

```diff
@@ -87,19 +88,17 @@ public sealed class MethodShouldBeNamedAccordingToSynchronicity : SonarDiagnosti
                 },
                 SyntaxKind.MethodDeclaration);
 
-        private static bool HasAsyncReturnType(IMethodSymbol methodSymbol)
-        {
-            var returnSymbol = (methodSymbol.ReturnType as INamedTypeSymbol)?.ConstructedFrom;
-            if (returnSymbol == null || returnSymbol.Is(KnownType.Void))
-            {
-                return false;
-            }
+        private static bool HasAsyncReturnType(IMethodSymbol methodSymbol) =>
+            methodSymbol.ReturnType is ITypeParameterSymbol typeParameter
+                ? typeParameter.ConstraintTypes.Any(x => IsAsyncType(x))
```

Comment:
What about `type` or `symbol` indead fo `x`?

Reply:
According to our [coding style](https://github.com/SonarSource/sonar-dotnet/blob/master/docs/coding-style.md#naming-conventions), we should use `x`.

Reply:
This is probably not the place to challenge the coding style guidelines. ;) But a short descriptive name for a variable could not harm, and could improve repeatability. That being said, the rest looks good to me.

Reply:
The `x` is an intentional convention based on lambda-x calculus. In simple cases, it helps to easily distinguish between the iterated variable vs. local fields/variables.

In this specific case, it could have been `Any(IsAsyncType)` instead.